cli/config/configuration-acrn.toml.in

```diff
@@ -16,6 +16,11 @@ ctlpath = "@ACRNCTLPATH@"
 kernel = "@KERNELPATH_ACRN@"
 image = "@IMAGEPATH@"
 
+# List of valid annotation names for the hypervisor
+# Each member of the list is a regular expression, which is the base name
+# of the annotation, e.g. "path" for io.katacontainers.config.hypervisor.path"
+enable_annotations = @DEFENABLEANNOTATIONS@
```

Comment:
This is quite a big change to disable annotations altogether. We should discuss this in the arch call and document this properly after agreement. Also annotations will no longer work for any users making use of this feature, so I think we need to highlight this once this goes in a release.

Reply:
> This is quite a big change to disable annotations altogether. We should discuss this in the arch call and document this properly after agreement. Also annotations will no longer work for any users making use of this feature, so I think we need to highlight this once this goes in a release.

@amshinde This change has been discussed in the arch meeting several times already. It was discussed since May. See https://bugs.launchpad.net/katacontainers.io/+bug/1878234 and https://github.com/kata-containers/runtime-private/pull/1 to see the extent of the discussion, to which you took place. I only moved here once it was approved. However, now that the cat is out of the bag, I would like to move ahead with what was approved, since this is a security risk and a CVE, and not restart here a discussion that was help under wraps for good reasons over the past 6 months.

Reply:
> Another question I have is I only see this for hypervisor specific section here, so are annotations related to the runtime/agent affected by this change?

No. Annotations in the agent cannot cause the kind of damage the runtime can. See the discussion on Launchpad.

Reply:
I just found it a bit inconsistent that the hypervisor related annotations are disabled while those related to runtime/agent  are enabled. Its worth making this clear in the commit message and in the PR itself.

Reply:
The plan is definitely to add similar control for other components. if only for consistency. However, since this does not represent a security risk, it can be deferred. We discussed that earlier.

Reply:
@amshinde, would you be okay with this change is as long as a patch updating our documentation comes together with this series?

Reply:
Yes, I am fine with it as long as we document it.

Reply:
@amshinde Could you clarify, when you talk about defaults, if you are talking about

1. The defaults from the binary if there is nothing in the configuration file  (currently empty list for everything)
2. The defaults in the `config*.toml.in` files, which currently are always some `@DEFBLAH`
3. The defaults in the makefile.

Right now, for example, the value of `DEFENABLEANNOTATIONS` in the makefile is `[]`. I could very easily be convinced to replace it with `[".*"]`, i.e. enable hypervisor annotations in our builds. Is that what you were talking about?

I'll push a commit with that single change in order to be able to forward-port it to 2.0. @kata-containers/architecture-committee , do you agree that enabling hypervisor annotations in the default configuration files is the right thing to do?

Reply:
@amshinde Assuming I (finally) correctly understood what you suggested, the change is in f26b3f9bf78fa12a448b3f98b5578748c2164a9d. I marked it as "suggested-by" you. If I misunderstood, let me know (and reopen this thread).